src/software/ai/navigator/path_planner/global_path_planner.h

```diff
@@ -0,0 +1,58 @@
+#include <map>
+#include <set>
+
+#include "enlsvg_path_planner.h"
+#include "shared/parameter/cpp_dynamic_parameters.h"
+#include "software/ai/hl/stp/tactic/tactic.h"
+#include "software/ai/motion_constraint/motion_constraint.h"
+#include "software/ai/motion_constraint/motion_constraint_set_builder.h"
+#include "software/ai/navigator/obstacle/robot_navigation_obstacle_factory.h"
+#include "software/geom/linear_spline2d.h"
+#include "software/geom/point.h"
+#include "software/logger/logger.h"
+
+using Path = LinearSpline2d;
+
+// TODO #2504: The GLobalPathPlanner doesn't handle
+// MotionConstraint::AVOID_BALL_PLACEMENT_INTERFERENCE and
+// MotionConstraint::HALF_METER_AROUND_BALL correctly in most situations
+/**
+ * GlobalPathPlannerFactory is a module that constructs every possible path planner for
+ * every possible combination of static obstacles. It pre-computes static obstacle
+ * planners in order to achieve path planning efficiencies during the game.
+ *
+ * The GlobalPathPlannerFactory uses MotionConstraints to create these static obstacles.
+ */
+class GlobalPathPlannerFactory
+{
+   public:
+    /**
+     * Creates path planners for every possible combination of obstacles using the given
+     * parameters.
+     *
+     * @param navigation_obstacle_config the config is used to convert motion constraints
+     * into obstacles
+     * @param world                      the world is used to create the path planner gand
+     * obstacles
+     * @param navigable_area             the path planner's navigable area
+     */
+    GlobalPathPlannerFactory(const std::shared_ptr<const RobotNavigationObstacleConfig>
+                                 navigation_obstacle_config,
+                             const World &world, const Rectangle &navigable_area);
```

Comment:
I think we should eliminate the navigable area since it can inconsistent with the field, which where obstacles are created

Reply:
updated